src/main/java/ch/njol/skript/conditions/CondAppliedEffect.java

```diff
@@ -0,0 +1,59 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.event.block.BeaconEffectEvent;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Applied Effect")
+@Description("Checks to see if the applied effect of a beacon effect event is primary or secondary")
+@Examples({
+	"on beacon effect:",
+		"\tif applied effect is primary:",
+			"\t\tbroadcast \"Is Primary\"",
+		"\telse if applied effect is secondary:",
+			"\t\tbroadcast \"Is Secondary\""
+})
+@RequiredPlugins("Paper")
+@Since("INSERT VERSION")
+public class CondAppliedEffect extends Condition {
+
+	static {
+		Skript.registerCondition(CondAppliedEffect.class,
+			"[the] applied effect is [the] (primary|:secondary) [[potion] effect]",
+			"[the] applied effect (isn't|is not) [the] (primary|:secondary) [[potion] effect]"
```

Comment:
Wouldn't this be better served as an event-value expression `applied [beacon] effect`? Then it'd just use CondCompare rather than needing a special condition.